include/pacemaker.h

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#ifndef PACEMAKER__H
+#  define PACEMAKER__H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \file
+ * \brief High Level API
+ * \ingroup pacemaker
+ */
+
+#  include <crm/stonith-ng.h>
+#  include <libxml/tree.h>
+
+/*!
+ * \brief Perform a STONITH action.
+ *
+ * \param[in] st        A connection to the STONITH API.
+ * \param[in] target    The node receiving the action.
+ * \param[in] action    The action to perform.
+ * \param[in] name      Who requested the fence action?
+ * \param[in] timeout   How long to wait for the operation to complete (in seconds).
+ * \param[in] tolerance If a successful action for \p target happened within
+ *                      this many seconds, return 0 without performing the
+ *                      action again.
+ *
+ * \return 0 on success, or various error codes on error.
+ */
+int pcmk_fence_action(stonith_t *st, const char *target, const char *action,
+                      const char *name, int timeout, int tolerance);
```

Comment:
I've been thinking about the use of stonith_t and stonith_key_value_t in the new API. I think overhauling the stonith IPC API first would be unreasonable at this point, so there's not much of a way around it. However I'm thinking we can #ifdef out the public API here for now, and not release it until we get around to the IPC API. I don't think that will be a problem since this API is mainly an initial demonstration rather than an urgent need.

Reply:
Yeah, that's reasonable.  Once everything else is looking good, I'll add a standalone patch that adds the ifdefs.  Hopefully later we can just git revert that single patch.